billboard.py

```diff
@@ -339,6 +375,51 @@ def getMeta(attribute, ifNoValue=None):
             )
             self.entries.append(entry)
 
+    def _parseYearEndPage(self, soup):
+        # Parse the chart's year from its URL
+        try:
+            href = soup.select_one("link").get("href")
+            self.year = str(re.search(r"/((1|2)\d{3})/", href).group(1))
```

Comment:
Just curious, does doing `\d{4}` not work for some reason?

Reply:
Oh ha, this was probably excessive, but I figured the only valid dates will start with either a 1 or 2. I'm fine with `/\d{4}/` if you'd prefer.